src/Clapi/Types/Digests.hs

```diff
@@ -56,7 +56,7 @@ trpDigest :: Seg -> TrpDigest
 trpDigest ns = TrpDigest ns mempty alEmpty mempty mempty
 
 trpdRemovedPaths :: TrpDigest -> [Path]
-trpdRemovedPaths (TrpDigest _ _ _ cOps _) = Map.foldlWithKey f [] cOps
+trpdRemovedPaths (TrpDigest ns _ _ cOps _) = (ns :</) <$> Map.foldlWithKey f [] cOps
```

Comment:
Was there a reason not to qualify them outside of this function? It's gone from being a kind of data projection function to doing some transformation...

Reply:
I was thinking that since it took the whole digest in that it should take into account whatever information the digest contained so the caller didn't have to unpack it. If you want the other one then I think this should take the `ContainerOps` not the digest since that would make explicit that the caller will need to have unpacked the digest not to end up unqualified.